rust/otap-dataflow/crates/otap/src/otap_batch_processor/metrics.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+//! Metrics for the OTAP batch processor.
+
+use otap_df_telemetry::instrument::Counter;
+use otap_df_telemetry_macros::metric_set;
+
+/// Minimal, essential metrics for the OTAP batch processor, plus observability for
+/// dirty-flag behavior and timer flush decisions.
+#[metric_set(name = "otap.processor.batch")]
+#[derive(Debug, Default, Clone)]
+pub struct OtapBatchProcessorMetrics {
```

Comment:
For the naming of the metrics, the collector defined this RFC. We should follow their recommendations for now. Example instead of `received_rows_logs`, I thing they recommend `log_consumed`. I let you check that in details.

https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/rfcs/component-universal-telemetry.md

Reply:
ok, sounds good, I missed that.